File: bee_slack_app/model/book.py
from typing import TypedDict


class Book(TypedDict):
    """
    本
    """

    isbn: str
    title: str
    author: str
    url: str
    image_url: str
    description: str
    updated_at: str


class RecommendBook(Book):
    """
    おすすめの本
    """

    interested: bool
    ml_model: str

File: src/model/book.ts
/**
 * 本
 */
export type Book = {
  isbn: string;
  title: string;
  author: string;
  url: string;
  imageUrl: string;
  description: string;
  updatedAt: string;
};

export type RecommendBook = Book & {
  interested: boolean;
  mlModel: string;
};

export type SearchedBook = {
  userId: string;
  isbn: string;
  mlModel: string;
  interested: boolean;
  updatedAt?: string;
};
